src/include/sof/audio/codec_adapter/codec/generic.h

```diff
@@ -25,6 +25,41 @@
 				(value)); \
 	} while (0)
 
+#define DECLARE_CODEC_ADAPTER(adapter, uuid, tr) \
+static struct comp_dev *adapter_shim_new(const struct comp_driver *drv, \
+					 struct sof_ipc_comp *comp)\
+{ \
+	return codec_adapter_new(drv, comp, adapter);\
+} \
+\
```

Comment:
Ok, I see where this is going. So, now we also need to modify codec_adapter_new to accommodate for the 3rd parameter. This is the way we tell the generic codec adapter which specific ops implementation to use.

Reply:
yes, and we can add another hook into the register routine to register the codec to a codec list which the multiplexer can pull from